Simplified/NYPLMigrationManager.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+/**
+Manages data migrations as they are needed throughout the app's life
+
+App version is cached in UserDefaults and last cached value is checked against current build version
+and updates are applied as required
+*/ 
+class MigrationManager: NSObject {
+  @objc static func migrate() {
+    // Fetch target version
+    let targetVersion = Bundle.main.infoDictionary!["CFBundleShortVersionString"] as! String
+
+    // Fetch and parse app version
+    let appVersion = NYPLSettings.shared().appVersion ?? ""
+    let appVersionTokens = appVersion.filter({ $0.isNumber || $0 == "." }).split(separator: ".").map({ Int($0)! })
```

Comment:
You have ensured that `Int($0)!` should never crash, so I think this is good functionally. This would be a little bit shorter/simpler though:
`appVersion.split(separator: ".").compactMap({ Int($0)? })`

Reply:
Oh, there is a functional difference though: my line would ignore any components that contain numbers AND non-numbers. I hope that we would only ever have numbers, but worth noting at least.